Libraries/CustomComponents/ListView/ListView.js

```diff
@@ -164,6 +164,13 @@ var ListView = React.createClass({
      */
     renderSectionHeader: PropTypes.func,
     /**
+     * (props) => renderable
+     *
+     * A function that returns the scroll view in which the list rows are
+     * rendered. Defaults to returning a ScrollView with the given props.
+     */
+    renderScrollView: React.PropTypes.func.isRequired,
```

Comment:
Oh, interesting, does flow even apply to non-es6 class-style propTypes?
On Thu, Jun 11, 2015 at 4:29 PM James Ide <notifications@github.com> wrote:

> In Libraries/CustomComponents/ListView/ListView.js
> <https://github.com/facebook/react-native/pull/785#discussion_r32279234>:
>
> > @@ -164,6 +164,13 @@ var ListView = React.createClass({
> >       */
> >      renderSectionHeader: PropTypes.func,
> >      /**
> > +     * (props) => renderable
> > +     *
> > +     * A function that returns the scroll view in which the list rows are
> > +     * rendered. Defaults to returning a ScrollView with the given props.
> > +     */
> > +    renderScrollView: React.PropTypes.func.isRequired,
>
> Will update with a renamed version.
>
> Regarding isRequired, I understand it to mean that the type of
> this.props.renderScrollView is Function instead of ?Function in
> Flow-speak. So because we're saying the prop is non-null (which is true
> because of the default value from getDefaultProps), Flow won't complain
> about this.props.renderScrollView() needing a null check.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/facebook/react-native/pull/785/files#r32279234>.
>